stan/math/prim/fun/atanh.hpp

```diff
@@ -36,12 +36,8 @@ inline double atanh(double x) {
  * @throw std::domain_error If argument is less than 1.
  */
 inline double atanh(int x) {
-  if (is_nan(x)) {
-    return x;
-  } else {
-    check_bounded("atanh", "x", x, -1, 1);
-    return std::atanh(x);
-  }
+  check_bounded("atanh", "x", x, -1, 1);
+  return std::atanh(x);
```

Comment:
Just a sanity check, this was changed because the standard says below right?

> If a domain error occurs, an implementation-defined value is returned (NaN where supported)

Reply:
Figure nan is a domain error

Reply:
This was changed because I feel that its an unnecessary check. An int cant be NaN as NaN is defined only for floating point numbers.

`std::isnan(x)` that is used underneath will just cast the int to double and do a check.

Reply:
Not a huge deal probably as I doubt many call atanh with ints.